auto_reconcile_structured_ref/models/account_bank_statement_line.py

```diff
@@ -0,0 +1,117 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, api
+from odoo.exceptions import UserError
+
+
+class AccountBankStatementLine(models.Model):
+    _inherit = "account.bank.statement.line"
+
+    @api.multi
+    def auto_reconcile(self):
+        # If we don't have esr_reconcile we want the default odoo way
+        if not self.env.context.get('struct_reconcile'):
+            return super(AccountBankStatementLine, self).auto_reconcile()
+        # otherwise use struct reconcile
+        else:
+            return self.auto_reconcile_struct()
+
+    @api.multi
+    def auto_reconcile_struct(self):
+        """ Reconcile the Bank statement line and acount move line
+        based on only the strctured reference
+
+        """
+        self.ensure_one()
+        match_recs = self.env['account.move.line']
+        company_currency = self.journal_id.company_id.currency_id
+        # customized: dropped unneeded params
+        precision = company_currency.decimal_places
+        params = {
+            'company_id': self.env.user.company_id.id,
+            'account_payable_receivable': (
+                self.journal_id.default_credit_account_id.id,
+                self.journal_id.default_debit_account_id.id
+            ),
+            'partner_id': self.partner_id.id,
+            'ref': self.name.strip(),
+            'amount': self.amount,
+            'precision': precision
+        }
+        # Try to get Structured Ref match
+        if self.name:
+            sql_query = self._get_common_sql_query_ignore_partner() + \
+                " AND POSITION(aml.ref in %(ref)s) > 0" \
+                " AND aml.ref is not null" \
+                " AND round(aml.amount_residual,%(precision)s)" \
+                " = round(%(amount)s,%(precision)s) " \
+                " ORDER BY" \
+                " date_maturity asc, aml.id asc"
+            self.env.cr.execute(sql_query, params)
+            match_recs = self.env.cr.dictfetchall()
+            if len(match_recs) > 1:
+                return False
+        # Customizations done. odoo code from here on.
+        if not match_recs:
+            return False
+
+        match_recs = self.env['account.move.line'].browse(
+            [aml.get('id') for aml in match_recs]
+        )
+        # Now reconcile
+        counterpart_aml_dicts = []
+        payment_aml_rec = self.env['account.move.line']
+        for aml in match_recs:
+            if aml.account_id.internal_type == 'liquidity':
+                payment_aml_rec = (payment_aml_rec | aml)
+            else:
+                amount = self.amount
```

Comment:
You indicate line 56 that the customizations are done, and back to odoo code.
But for amount, I see that on Odoo: 
`amount = aml.currency_id and aml.amount_residual_currency or aml.amount_residual`
https://github.com/odoo/odoo/blob/10.0/addons/account/models/account_bank_statement.py#L720

Reply:
good catch. im not sure why this change is there ill update it

Reply:
updated.